lib/overcommit/hook/pre_commit/line_endings.rb

```diff
@@ -0,0 +1,51 @@
+module Overcommit::Hook::PreCommit
+  # Checks for line endings in files.
+  class LineEndings < Base
+    def run
+      messages = []
+
+      offending_files.map do |file_name|
+        file = File.open(file_name)
+        file.each_line do |line|
+          # remove configured line-ending
+          line.gsub!(/#{config['eol']}/, '')
+
+          # detect any left over line-ending characters
+          next unless line.end_with?("\n", "\r")
+
+          messages << Overcommit::Hook::Message.new(
+            :error,
+            file_name,
+            file.lineno,
+            "#{file_name}:#{file.lineno}:#{line.inspect}"
+          )
+        end
+      end
+
+      messages
+    end
+
+    private
+
+    def offending_files
+      result = execute(%w[git ls-files --eol -z --], args: applicable_files)
+      raise 'Unable to access git tree' unless result.success?
+
+      result.stdout.split("\0").map do |file_info|
+        i, _w, _attr, path = file_info.split
+        next if i == 'l/-text' # ignore binary files
+        next if i == "l/#{eol}"
+        path
+      end.compact
+    end
+
+    def eol
+      @eol ||=  case config['eol']
+                when "\n"
+                  'lf'
+                when "\r\n"
+                  'crlf'
+                end
```

Comment:
We should handle the case when the user specifies an unsupported line ending, rather than failing silently and return `nil`.